controllers/componentsubscription_controller.go

```diff
@@ -287,3 +289,83 @@ func (r *ComponentSubscriptionReconciler) reconcile(ctx context.Context, obj *v1
 	// Always requeue to constantly check for new versions.
 	return ctrl.Result{RequeueAfter: obj.GetRequeueAfter()}, nil
 }
+
+func (r *ComponentSubscriptionReconciler) reconcileMpasComponent(
+	ctx context.Context,
+	octx ocm2.Context,
+	obj *v1alpha1.ComponentSubscription,
+	sourceComponentVersion ocm2.ComponentVersionAccess,
+) error {
+	if obj.Spec.Destination == nil {
+		return fmt.Errorf("destination must be set for MPAS enabled components")
+	}
+
+	rreconcile.ProgressiveStatus(false, obj, meta.ProgressingReason, "transferring mpas component to target repository: %s", obj.Spec.Destination.URL)
+
+	if err := r.checkComponentIsMPASEnabled(sourceComponentVersion); err != nil {
+		return fmt.Errorf("failed to verify component validity: %w", err)
+	}
+
+	pub, err := r.OCMClient.SignDestinationComponent(ctx, sourceComponentVersion)
```

Comment:
it's not the component that we need to sign. it's the digest of the spec of the subscription. otherwise what's the point in having the digest?